### ISSUE TITLE
Add lint for calling `VecDeque::swap` as it might panic

### ISSUE THREAD

**Issue by austenadler** [rust-lang/rust-clippy#7553]
### What it does

Triggers on use of possibly-panicking function [`VecDeque::swap`](https://doc.rust-lang.org/std/collections/struct.VecDeque.html#method.swap) when index is out of bounds, similar to existing lint `clippy::indexing_slicing`.

### Categories (optional)

- Kind: restriction

Recommended code would not panic, swap would. For users trying to prevent all (or all non-std) panics, this would be useful.

For example:
- Do not use `.swap`, instead use `let (a, b) = get_mut(0)?, get_mut(1)?; std::mem::swap(&a, &b);`

### Drawbacks

None.

### Example

```rust
#![warn(clippy::all, clippy::pedantic, clippy::nursery, clippy::restriction)]
#![allow(clippy::missing_docs_in_private_items, clippy::blanket_clippy_restriction_lints)]
use std::collections::VecDeque;

fn main() {
    let mut c: VecDeque<u8> = VecDeque::new();
    c.push_back(0);
    c.push_back(1);
    
    let _ = c[0]; // clippy::indexing_slicing correctly detects possible panic

    c.swap(0, 1); // Could cause panic; clippy ignores this line
    // c.swap(1, 2); // Example swap panic
}
```

Could be written as:

```rust
// ...
  // Honestly I don't know the right way to do this. I would expect something like this
  // But you can't borrow c mutably more than once
  let (a, b) = (c.get_mut(0)?, c.get_mut(1)?);
  std::mem::swap(&a, &b);
// ...
```
